src/data/reports/data-quality/DataQualityDefaultRepository.ts

```diff
@@ -0,0 +1,276 @@
+                            expression: true,
+                            filter: true,
+                        },
+                        filter: {
+                            lastUpdated: {
+                                gt: dataQuality?.programIndicatorsLastUpdated,
+                            },
+                        },
+                    },
+                })
+                .getData();
+
+            await promiseMap(indicators, async indicator => {
+                const dataQualityItem: DataQualityItem = indicator;
+                try {
+                    const numeratorValidation = await this.api.expressions
+                        .validate("indicator", indicator.numerator)
+                        .getData();
+                    const denominatorValidation = await this.api.expressions
+                        .validate("indicator", indicator.denominator)
+                        .getData();
+
+                    if (numeratorValidation.status === "ERROR" || denominatorValidation.status === "ERROR") {
+                        dataQualityItem.numeratorresult = numeratorValidation.status === "ERROR";
+                        dataQualityItem.denominatorresult = denominatorValidation.status === "ERROR";
+                        dataQualityItem.metadataType = "Indicator";
+                        dataQualityErrors.push(dataQualityItem);
+                    }
+                } catch (error) {
+                    console.debug(error);
+                }
+            });
+            await promiseMap(programIndicators, async programIndicator => {
+                const dataQualityItem: DataQualityItem = programIndicator;
+                try {
+                    const expressionValidation = await this.api.expressions
+                        .validate("program-indicator-formula", programIndicator.expression)
+                        .getData();
+                    const filterValidation = await this.api.expressions
+                        .validate("program-indicator-filter", programIndicator.filter)
+                        .getData();
+
+                    if (expressionValidation.status === "ERROR" || filterValidation.status === "ERROR") {
+                        dataQualityItem.numeratorresult = expressionValidation.status === "ERROR";
+                        dataQualityItem.denominatorresult = filterValidation.status === "ERROR";
+                        dataQualityItem.metadataType = "ProgramIndicator";
+                        dataQualityErrors.push(dataQualityItem);
+                    }
+                } catch (error) {
+                    console.debug(error);
+                }
+            });
+
+            await this.saveDataQuality(Namespaces.DATA_QUALITY, {
+                indicatorsLastUpdated: new Date().toISOString(),
+                programIndicatorsLastUpdated: new Date().toISOString(),
+                validationResults: dataQualityErrors,
+            });
```

Comment:
This means that we only store the items newer than `dataQuality.programIndicatorsLastUpdated`, discarding the previous content.